lite/kernels/imagination_nna/bridges/conv_op.cc

```diff
@@ -0,0 +1,278 @@
+                                                 filter_dims.Vectorize(),
+                                                 qnt,
+                                                 Node::Role::kConst);
+  imgdnn_tensor filter_tensor = filter_node->data();
+
+  // Add bias node if exists bias
+  // Supports the bias nodes with the following dimensions
+  // 0: {oc}
+  // 1: {1, oc, oh, ow}
+  // 2: {n, oc, oh, ow}
+  std::shared_ptr<Node> bias_node = NULL;
+  imgdnn_tensor bias_tensor = nullptr;
+  if (HasInputArg(op_info, scope, "Bias")) {
+    auto bias_name = op_info->Input("Bias").front();
+    if (graph->Has(bias_name)) {
+      bias_node = graph->Get(bias_name);
+    } else {
+      auto bias = scope->FindMutableTensor(bias_name);
+      auto bias_dims = bias->dims();
+      auto bias_data_size = bias_dims.production();
+      auto output_data_size = output_dims.production();
+      std::vector<int64_t> bias_shape;
+      if (bias_data_size == oc) {
+        // 0: {oc}
+        bias_shape = {1, oc, 1, 1};
+      } else if (bias_data_size == output_data_size / bs) {
+        // 1: {1, oc, oh, ow}
+        bias_shape = {1, output_dims[1], output_dims[2], output_dims[3]};
+      } else if (bias_data_size == output_data_size) {
+        // 2: {n, oc, oh, ow}
+        bias_shape = output_dims.Vectorize();
+      } else {
+        LOG(WARNING)
+            << "[NNA] Bias dimension " << bias_dims
+            << " isn't supported in conv2d Op when output dimension is "
+            << output_dims;
+        return FAILED;
+      }
+
+      TensorInfoReset(&qnt);
+      qnt.type = IMGDNN_TYPE_I32;
+      if (per_channel) {
+        qnt.scales.resize(bias_data_size);
+        for (int i = 0; i < bias_data_size; i++)
+          qnt.scales[i] = input_scale * weight_scale[i];
+        qnt.zero_points.assign(bias_data_size, 0);
+        qnt.count = 2;
+        qnt.axis = 1;
+      } else {
+        qnt.scales.push_back(input_scale * weight_scale[0]);
+        qnt.zero_points.push_back(0);
+      }
+
+      int quant_bits = 32;
+      auto dtype_max = static_cast<int>((1 << (quant_bits - 1)) - 1);
+      auto dtype_min = static_cast<int>(0 - dtype_max);
+
+      auto bias_data = bias->data<float, float>();
+      int32_t *bias_qnt_data =
+          reinterpret_cast<int32_t *>(graph->GetBuilder()->GetBufromPool(
+              bias_dims.production() * sizeof(int32_t)));
+      for (int i = 0; i < bias_data_size; i++) {
+        float current_scale = per_channel ? qnt.scales[i] : qnt.scales[0];
+        bias_qnt_data[i] = std::min(
+            std::max(static_cast<int>(bias_data[i] / current_scale), dtype_min),
+            dtype_max);
+      }
+
+      std::vector<int64_t> shapes{1, oc};
+      bias_node =
+          graph->Add(bias_name, bias_qnt_data, shapes, qnt, Node::Role::kConst);
+
+      bias_tensor = bias_node->data();
+    }
+  }
+
+  unsigned int img_stride[2] = {(unsigned int)strides[0],
```

Comment:
use C++ style `static_cast` to cast the type here.

Reply:
Done